core/src/test/java/de/jplag/RootFolderTest.java

```
@@ -0,0 +1,73 @@
+        JPlagResult result = runJPlag(paths, it -> it);
+        assertEquals(5, result.getNumberOfSubmissions());
+    }
+
+    @Test
+    void testMultiRootDirSeparateBasecode() throws ExitException {
+        String basecodePath = getBasePath(GLOBAL_BASECODE);
+        List<String> paths = List.of(getBasePath(BASE_SAMPLE), getBasePath(SIMPLE_SAMPLE)); // 3 + 2 submissions.
+        JPlagResult result = runJPlag(paths, it -> it.withBaseCodeSubmissionDirectory(new File(basecodePath)));
+        assertEquals(5, result.getNumberOfSubmissions());
+    }
+
+    @Test
+    void testMultiRootDirBasecodeInSubmissionDir() throws ExitException {
+        String basecodePath = getBasePath(BASE_SAMPLE, BASECODE_SUBMISSION);
+        List<String> paths = List.of(getBasePath(BASE_SAMPLE), getBasePath(SIMPLE_SAMPLE)); // 2 + 2 submissions.
+        JPlagResult result = runJPlag(paths, it -> it.withBaseCodeSubmissionDirectory(new File(basecodePath)));
+        assertEquals(4, result.getNumberOfSubmissions());
+    }
+
+    @Test
+    void testDisjunctNewAndOldRootDirectories() throws ExitException {
+        List<String> newDirectories = List.of(getBasePath(SIMPLE_SAMPLE)); // 2 submissions
+        List<String> oldDirectories = List.of(getBasePath(BASE_SAMPLE)); // 3 submissions
+        JPlagResult result = runJPlag(newDirectories, oldDirectories, it -> it);
+        int numberOfExpectedComparison = 1 + 3 * 2;
+        assertEquals(numberOfExpectedComparison, result.getAllComparisons().size());
+    }
+
+    @Test
+    void testOverlappingNewAndOldDirectoriesOverlap() throws ExitException {
+        List<String> newDirectories = List.of(getBasePath(SIMPLE_SAMPLE)); // 2 submissions
+        List<String> oldDirectories = List.of(getBasePath(SIMPLE_SAMPLE));
+        JPlagResult result = runJPlag(newDirectories, oldDirectories, it -> it);
+        int numberOfExpectedComparison = 1;
+        assertEquals(numberOfExpectedComparison, result.getAllComparisons().size());
+    }
+
```

Not necessary for this PR, but should overlapping old and new directories not rather throw an error?